src/main/java/ch/njol/skript/aliases/ItemData.java

```diff
@@ -178,68 +176,64 @@ public ItemStack getRandom() {
 			return new ItemStack(type, 1, (short) (Utils.random(dataMin, dataMax + 1)));
 		}
 	}
-	
+
 	public Iterator<ItemStack> getAll() {
 		if (typeid == -1) {
 			return new Iterator<ItemStack>() {
-				
 				@SuppressWarnings("null")
 				private final Iterator<Material> iter = Arrays.asList(Material.values()).listIterator(1); // ignore air
-				
+
 				@Override
 				public boolean hasNext() {
 					return iter.hasNext();
 				}
-				
+
 				@Override
 				public ItemStack next() {
 					return new ItemStack(iter.next(), 1);
 				}
-				
+
 				@Override
 				public void remove() {
 					throw new UnsupportedOperationException();
 				}
-				
 			};
 		}
 		if (dataMin == dataMax)
-			return new SingleItemIterator<ItemStack>(new ItemStack(typeid, 1, dataMin == -1 ? 0 : dataMin));
+			return new SingleItemIterator<>(new ItemStack(typeid, 1, dataMin == -1 ? 0 : dataMin));
 		return new Iterator<ItemStack>() {
-			
 			private short data = dataMin;
-			
+
 			@Override
 			public boolean hasNext() {
 				return data <= dataMax;
 			}
-			
+
 			@Override
 			public ItemStack next() {
 				if (!hasNext())
 					throw new NoSuchElementException();
 				return new ItemStack(typeid, 1, data++);
 			}
-			
+
 			@Override
 			public void remove() {
 				throw new UnsupportedOperationException();
 			}
-			
 		};
 	}
-	
+
+	@SuppressWarnings("MethodDoesntCallSuperMethod")

```

Comment:
Not standard warning. Not needed.

Reply:
Ok I guess that warning I'll just disable for the whole project, I don't see how this warning can really useful in places, excluding than maybe Android projects tbf (unless clone() really should always call the super method and Njol used it wrong)

Reply:
Clone does not need to call super.clone() as long as it produces a copy. What kind of copy, is not defined; if it is not a deep copy, Javadoc should probably mention it. But even that is optional.